ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -390,16 +390,6 @@ public void commit (boolean report_heuristics) throws HeuristicMixed,
 			}
 
 			destroyAction();
-		} else if (outcome == ActionStatus.ABORTED || outcome == ActionStatus.H_ROLLBACK) {
-			throw new TRANSACTION_ROLLEDBACK(ExceptionCodes.FAILED_TO_COMMIT,
```

Comment:
We are normally very conservative about changing behaviour in the JTS code since it has evolved over many years to address many edge cases.

You have not demonstrated any actual issue with the commit code and in fact I see that you had a test failure in  the commit and had to modify two tests (TransactionTest1 and TerminationTest) to make it work. So unless there are good reasons behind a change we normally reject them.